sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1943,7 +1944,11 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
               Row(Seq(Row(1, 2)), Seq(Map(false -> "def", true -> "jkl"))),
               Seq(Map(true -> "xyz"))),
             Row(2,
-              null,
+              if (config.dataSource != "orc") {
```

Comment:
Thank you for review, @dtenedor .
- Please see https://issues.apache.org/jira/browse/SPARK-41782 . We have a benchmark to detect this kind of perf regression. You can run it locally in your environment.
- This is a partial revert to the original code which is the existing behavior before your PR like the previous Spark. As I mentioned in the PR description, [SPARK-39862](https://issues.apache.org/jira/browse/SPARK-39862) should propose a fix without perf regression.

New feature is good as long as not breaking the old behavior.

Reply:
> Thank you for review, @dtenedor .
> 
> * Please see https://issues.apache.org/jira/browse/SPARK-41782 . We have a benchmark to detect this kind of perf regression. You can run it locally in your environment.

Thanks @dongjoon-hyun for the benchmark! The Jira simply comprises the title `Regenerate benchmark results`. Is there some instructions for how to run the benchmark?

> * This is a partial revert to the original code which is the existing behavior before your PR like the previous Spark. As I mentioned in the PR description, [SPARK-39862](https://issues.apache.org/jira/browse/SPARK-39862) should propose a fix without perf regression.
> 
> New feature is good as long as not breaking the old behavior.

Agree on this. However, that bug fix was merged into Spark 3.3 on Jul. 28, 2022. Is it possible that users could have built pipelines since then using the new feature that would return incorrect results if we merged this PR?

Reply:
If someone can tell me how to run the benchmark, I can play around with a fix for the perf regression that also does not change the behavior. I suspect that it's due to the extra function call overhead that takes place each time a value is written, but not sure without a profile :)

Reply:
I copied a wrong link. Here is the exact link, @dtenedor . 
https://github.com/apache/spark/pull/39301#discussion_r1059239575

Reply:
Here is the benchmark.
- https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala

Reply:
Please run the above command and check the following case specifically, @dtenedor . 

https://github.com/apache/spark/blob/ec594236df4a568d3b844089b471373789a67b2a/sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala#L400

Reply:
@dongjoon-hyun sounds good, the comment in that file explains how to run it. Let me take a look myself, I will try to fix the performance regression without changing the behavior. I can ping this thread again with the results

Reply:
Thanks, but I filed SPARK-41858 as the blocker of Apache Spark 3.4 because this really blocks Apache Spark 3.4 preparation from my side. If you don't mind, I'd prefer to merge this first and help you cleanly.

Reply:
@dongjoon-hyun that is OK, but we should probably not release Spark 3.4 with a correctness regression either. It is equally important as performance. If we create another Jira blocking the release of 3.4 that covers fixing the correctness bug, it is fine to merge this PR.

At any rate, I am hoping to figure this out today. Then we should be unblocked 🤔

Reply:
Thank you so much for your understanding. I'll file another blocker JIRA for that of course.

Reply:
LGTM, let's merge this then

Reply:
You can assign the new correctness Jira for me to fix.